### PR TITLE
Use Python type name instead of Spark's in error messages.

### PR DESCRIPTION
**Pull request by ueshin**
Addressing https://github.com/databricks/koalas/pull/1980#discussion_r547607957 to add pandas dtypes.

### CODE REVIEW

File: databricks/koalas/typedef/typehints.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def as_spark_type(tpe) -> types.DataType:
     """
     # TODO: Add "boolean" and "string" types.
     # ArrayType
-    if tpe in (np.ndarray,):
+    if tpe in (list, np.ndarray,):
```
Comment (ueshin) [inline databricks/koalas/typedef/typehints.py:106]: Is this reasonable?

Comment (HyukjinKwon) [reply databricks/koalas/typedef/typehints.py:106]: Yeah, I think it's fine.

File: databricks/koalas/typedef/typehints.py [hunk 2]
```diff
@@ -142,7 +142,46 @@ def as_spark_type(tpe) -> types.DataType:
         raise TypeError("Type %s was not understood." % tpe)
 
 
-def spark_type_to_pandas_dtype(spark_type):
+def spark_type_to_python_type(spark_type: types.DataType) -> type:
+    """ Return the given Spark DataType to Python type. """
+    # ArrayType
+    if isinstance(spark_type, types.ArrayType):
+        return list
```
Comment (HyukjinKwon) [inline databricks/koalas/typedef/typehints.py:149]: Maybe `numpy.ndarray` instead of `list`.. ?

File: databricks/koalas/typedef/typehints.py [hunk 3]
```diff
@@ -142,7 +142,46 @@ def as_spark_type(tpe) -> types.DataType:
...
+    elif isinstance(spark_type, types.StringType):
+        return str
+    # TimestampType
+    elif isinstance(spark_type, types.TimestampType):
+        return datetime.datetime
+    else:
+        raise TypeError("Type %s was not understood." % spark_type.simpleString())
+
+
+def spark_type_to_pandas_dtype(spark_type: types.DataType) -> np.dtype:
```
Comment (HyukjinKwon) [inline databricks/koalas/typedef/typehints.py:184]: @ueshin, btw what about we just use `spark_type_to_pandas_dtype` in the error message?

Comment (ueshin) [reply databricks/koalas/typedef/typehints.py:184]: We can use it, but in that case, it will show only `object` for most cases.

Comment (HyukjinKwon) [reply databricks/koalas/typedef/typehints.py:184]: Hm .. what about we show both pandas dtype and spark type? for example ...

```
Could not convert pandas type (spark type) to numeric?
```

Comment (ueshin) [reply databricks/koalas/typedef/typehints.py:184]: Sure.

**Review by HyukjinKwon — APPROVED** [review]

**Pull request merged**